eng/pipelines/templates/stages/vmr-build-with-join.yml

```diff
@@ -0,0 +1,174 @@
+    os: linux
+
+stages:
+# Insert all stages that build verticals
+# and publish artifacts for each vertical
+- ${{ each stage in parameters.verticalsStages }}:
+  - ${{ insert }}: ${{ stage }}
+- ${{ if parameters.finalJoinCondition }}:
+  # This stage should contain all jobs that take inputs from all verticals and determine the exact final outputs.
+  - stage: VMR_Final_Join
+    displayName: 'VMR Final Join'
+    dependsOn:
+    - ${{ each stage in parameters.verticalsStages }}:
+      - ${{ if ne(stage.templateContext.validationOnly, 'true') }}:
+        - ${{ stage.stage }}
+    variables:
+    - ${{ if parameters.isBuiltFromVmr }}:
+      - name: vmrPath
+        value: $(Build.SourcesDirectory)
+    - ${{ else }}:
+      - name: vmrPath
+        value: $(Agent.BuildDirectory)/vmr
+    - template: ../variables/vmr-build.yml
+      parameters:
+        vmrBranch: ${{ parameters.vmrBranch }}
+    jobs:
+      - job: FinalJoin
+        displayName: Final Build Pass
+        pool: ${{ parameters.pool_Windows }}
+        timeoutInMinutes: 240
+        templateContext:
+          outputParentDirectory: $(Build.ArtifactStagingDirectory)/artifacts
+          inputs:
+            # Download the artifacts from every job in every vertical stage under VerticalArtifacts
+            - ${{ each stage in parameters.verticalsStages }}:
+              - ${{ if ne(stage.templateContext.validationOnly, 'true') }}:
+                - ${{ each vertical in stage.jobs }}:
+                  - input: pipelineArtifact
+                    path: $(Build.ArtifactStagingDirectory)/VerticalArtifacts/${{ vertical.job }}
+                    artifactName: ${{ vertical.job }}_Artifacts
+            - input: pipelineArtifact
+              path: $(Build.ArtifactStagingDirectory)/VerticalManifests
+              artifactName: VerticalManifests
+          outputs:
+          - output: pipelineArtifact
+            path: $(Build.ArtifactStagingDirectory)/artifacts/MergedManifest.xml
+            artifact: AssetManifests
+            displayName: Publish Merged Manifest
+            sbomEnabled: false
+          - output: buildArtifacts
+            PathtoPublish: $(Build.ArtifactStagingDirectory)/artifacts/assets
+            ArtifactName: BlobArtifacts
+            displayName: Publish Blob Artifacts
+            sbomEnabled: false
+          - output: buildArtifacts
+            PathtoPublish: $(Build.ArtifactStagingDirectory)/artifacts/packages
+            ArtifactName: PackageArtifacts
+            displayName: Publish Package Artifacts
+            sbomEnabled: false
+          - output: pipelineArtifact
+            path: $(Build.ArtifactStagingDirectory)/artifacts/log
+            artifact: JoinVerticals_BuildLogs_Attempt$(System.JobAttempt)
+            displayName: Publish Build Logs
+            sbomEnabled: false
+            condition: succeededOrFailed()
+        steps:
+        - ${{ if ne(variables['System.TeamProject'], 'internal') }}:
```

Comment:
Only under non-internal?

Reply:
In internal, we use the 1ES PT inputs above.

I'd love to use those everywhere, but 1ES hasn't responded to my feature request for an "OSS 1ES PT with only the syntax niceties and no internal product validation"

Reply:
We could consider adding some basic inputs/outputs support to an Arcade template so we could at least unify the many places Arcade and other repos have to write the different styles.